src/fastcs/backends/graphQL/graphQL.py

```diff
@@ -0,0 +1,225 @@
+
+        uvicorn.run(
+            self._app,
+            host=options.host,
+            port=options.port,
+            log_level=options.log_level,
+        )
+
+
+def _wrap_attr_set(
+    d_attr_name: str,
+    attribute: AttrW[T],
+) -> Callable[[T], Coroutine[Any, Any, None]]:
+    async def _dynamic_f(value):
+        await attribute.process(value)
+        return value
+
+    # Add type annotations for validation, schema, conversions
+    _dynamic_f.__name__ = d_attr_name
+    _dynamic_f.__annotations__["value"] = attribute.datatype.dtype
+    _dynamic_f.__annotations__["return"] = attribute.datatype.dtype
+
+    return _dynamic_f
+
+
+def _wrap_attr_get(
+    d_attr_name: str,
+    attribute: AttrR[T],
+) -> Callable[[], Coroutine[Any, Any, Any]]:
+    async def _dynamic_f() -> Any:
+        return attribute.get()
+
+    _dynamic_f.__name__ = d_attr_name
+    _dynamic_f.__annotations__["return"] = attribute.datatype.dtype
+
+    return _dynamic_f
+
+
+def _wrap_as_field(
+    field_name: str,
+    strawberry_type: type,
+) -> StrawberryField:
+    def _dynamic_field():
+        return strawberry_type()
+
+    _dynamic_field.__name__ = field_name
+    _dynamic_field.__annotations__["return"] = strawberry_type
+
+    return strawberry.field(_dynamic_field)
+
+
+class NodeNotFoundError(Exception):
+    pass
+
+
+class FieldsTree:
+    def __init__(self, name: str):
+        self.name = name
+        self.children: list[FieldsTree] = []
+        self.fields_dict: dict[str, list[StrawberryField]] = {
+            "query": [],
+            "mutation": [],
+        }
+
+    def insert(self, path: list[str]) -> "FieldsTree":
+        # Create child if not exist
+        name = path.pop(0)
+        if self.is_child(name):
+            child = self.get_child(name)
+        else:
+            child = FieldsTree(name)
+            self.children.append(child)
+
+        # Recurse if needed
+        if path:
+            return child.insert(path)  # type: ignore
+        else:
+            return child
+
+    def is_child(self, name: str) -> bool:
+        for child in self.children:
+            if child.name == name:
+                return True
+        return False
+
+    def get_child(self, name: str) -> "FieldsTree":
+        for child in self.children:
+            if child.name == name:
+                return child
+        raise NodeNotFoundError
+
+    def create_type(self, strawberry_type: str) -> type | None:
+        for child in self.children:
+            if new_type := child.create_type(strawberry_type):
+                child_field = _wrap_as_field(
+                    child.name,
+                    new_type,
+                )
+                self.fields_dict[strawberry_type].append(child_field)
+
+        if self.fields_dict[strawberry_type]:
+            return create_type(
+                f"{self.name}{strawberry_type}", self.fields_dict[strawberry_type]
+            )
+        else:
+            return None
```

Comment:
I am quite confused by this method. I don't understand when it would return `None`. Isn't `strawberry_type` always just `"query"` or `"mutation"`?